noir-projects/aztec-nr/aztec/src/test/mocks/mock_note.nr

```diff
@@ -20,7 +20,22 @@ pub(crate) struct MockNote {
     pub(crate) value: Field,
 }
 
-impl NullifiableNote for MockNote {
+impl NoteInterface for MockNote {
+    fn get_note_type_id() -> Field {
+        // randomly chosen note type id --> has to fit within 7 bits
+        76
+    }
+}
+
+impl NoteHashing for MockNote {
+    fn compute_note_hash(self: Self, storage_slot: Field) -> Field {
+        // We use Poseidon2 instead of multi-scalar multiplication (MSM) here since this is not a partial note
+        // and therefore does not require MSM's additive homomorphism property. Additionally, Poseidon2 uses fewer
+        // constraints.
```

Comment:
```suggestion
```

MSM will be the exception, not the rule. We actually won't even use MSM until we have multivariant partials, which will likely not happen until at least Q4.